src/main/java/org/jabref/logic/importer/fetcher/SemanticScholar.java

```diff
@@ -0,0 +1,267 @@
+     * @throws IOException          if one of the URL is not correct
+     */
+    @Override
+    public Optional<URL> findFullText(BibEntry entry) throws IOException {
+        Objects.requireNonNull(entry);
+
+        Optional<DOI> doi = entry.getField(StandardField.DOI).flatMap(DOI::parse);
+        Optional<ArXivIdentifier> arXiv = entry.getField(EPRINT).flatMap(ArXivIdentifier::parse);
+
+        Document html = null;
+        if (doi.isPresent()) {
+            try {
+                // Retrieve PDF link
+                String source = SOURCE_ID_SEARCH + doi.get().getDOI();
+                html = Jsoup.connect(getURLBySource(source))
+                        .userAgent(URLDownload.USER_AGENT)
+                        .referrer("https://www.google.com")
+                        .ignoreHttpErrors(true)
+                        .get();
+            } catch (IOException e) {
+                LOGGER.info("Error for pdf lookup with DOI");
+            }
+        }
+        if (arXiv.isPresent() && entry.getField(EPRINT).isPresent()) {
+            // Check if entry is a match
+            String arXivString = entry.getField(EPRINT).get();
+            if (!arXivString.startsWith("arXiv:")) {
+                arXivString = "arXiv:" + arXivString;
+            }
+            String source = SOURCE_ID_SEARCH + arXivString;
+            html = Jsoup.connect(getURLBySource(source))
+                    .userAgent(URLDownload.USER_AGENT)
+                    .referrer("https://www.google.com")
+                    .ignoreHttpErrors(true)
+                    .get();
+        }
+        if (html == null) {
+            return Optional.empty();
+        }
+
+        // Retrieve PDF link from button on the webpage
+        // First checked is a drop-down menu, as it has the correct URL if present
+        // Else take the primary button
+        Elements metaLinks = html.getElementsByClass("flex-item alternate-sources__dropdown");
+        String link = metaLinks.select("a").attr("href");
+        if (link.length() < 10) {
+            metaLinks = html.getElementsByClass("flex-paper-actions__button--primary");
+            link = metaLinks.select("a").attr("href");
+        }
+        LOGGER.info("Fulltext PDF found @ SemanticScholar.");
+        return Optional.of(new URL(link));
+    }
+
+    @Override
+    public TrustLevel getTrustLevel() {
+        return TrustLevel.META_SEARCH;
+    }
+
+    /**
+     * Finds the URL for the webpage to look for the link to download the pdf.
+     *
+     * @param source the API link that contains the relevant information.
+     * @return the correct URL
+     * @throws MalformedURLException if error by downloading the page
+     */
+    public String getURLBySource(String source) throws IOException {
+        URLDownload download = new URLDownload(source);
+        JSONObject json = new JSONObject(download.asString());
+        LOGGER.debug(json.get("url").toString());
+        return json.get("url").toString();
```

Comment:
I think this could potentially lead to an NullPointerException if for some reason the json object has no field "url".